impl_core/src/test/java/io/opencensus/implcore/tags/propagation/TagContextRoundtripTest.java

```diff
@@ -48,13 +52,37 @@
   private final Tagger tagger = tagsComponent.getTagger();
 
   @Test
-  public void testRoundtripSerialization() throws Exception {
+  public void testRoundtripSerialization_NormalTagContext() throws Exception {
     testRoundtripSerialization(tagger.empty());
     testRoundtripSerialization(tagger.emptyBuilder().put(K1, V1).build());
     testRoundtripSerialization(tagger.emptyBuilder().put(K1, V1).put(K2, V2).put(K3, V3).build());
     testRoundtripSerialization(tagger.emptyBuilder().put(K1, V_EMPTY).build());
   }
 
+  @Test
+  public void testRoundtrip_TagContextWithMaximumSize() throws Exception {
+    TagContextBuilder builder = tagger.emptyBuilder();
+    int i = 0;
+    ByteArrayDataOutput byteArrayDataOutput = ByteStreams.newDataOutput();
+
+    // This loop should fill in tags that have a total size of 8185
+    while (byteArrayDataOutput.toByteArray().length
```

Comment:
It might be simpler to check the length of `serializer.toByteArray(builder.build())` on every iteration.  Then you wouldn't need to expose SerializationUtils.encodeTag.

Reply:
That's right, updated.